neotest-kotlin/app/src/main/kotlin/io/github/codymikol/neotestkotlin/framework/TestNode.kt

```diff
@@ -0,0 +1,182 @@
+package io.github.codymikol.neotestkotlin.framework
+
+import io.kotest.core.test.TestResult
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.JsonClassDiscriminator
+import kotlin.time.Duration
+
+/**
+ * A test node, the parent union type of a [TestNode.Test] and a [TestNode.Container]
+ * where a [TestNode.Test] is a terminal and a [TestNode.Container] can contain multiple
+ * other namespaces or tests.
+ */
+@OptIn(ExperimentalSerializationApi::class) // JsonClassDiscriminator is an experimental annotation for now
+@JsonClassDiscriminator("type")
+@Serializable
```

Comment:
I'm using Kotlinx serialization here, which is nice and all, but have some caveats from good ole Jackson. Namely, the fields in `TestNode.Container` for `status` and `duration` aren't evaluated because they're not backed fields. I'll probably pivot away from this if there's no way to handle this.

[docs](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/basic-serialization.md#backing-fields-are-serialized)